Add StraightVacateTask & rename existing to VacateAndReadjudicationTask

jcq opened the pull request:
Connects #12793

### Description
Rename StraightVacateAndReadjudication to VacateAndReadjudication, and implement new StraightVacate task

### Acceptance Criteria
- [ ] Code compiles correctly

jcq commented:
## Updated Screen Shot
<img width="557" alt="Screen Shot 2019-11-27 at 16 27 24" src="https://user-images.githubusercontent.com/2581274/69767821-e5d1f280-1132-11ea-9e5d-a245a640bb30.png">

tomas-nava commented on app/models/tasks/straight_vacate_task.rb:
```diff
@@ -0,0 +1,11 @@
+
+  def self.org(user)
+    JudgeTeam.for_judge(user.reload)
+  end
```
Not a blocker, but [we're trying to standardize class method definition](https://github.com/department-of-veterans-affairs/caseflow/issues/11778) using `class << self` blocks instead of `def self.method_name`.

jcq replied:
No prob — updated those!

tomas-nava commented on app/models/tasks/straight_vacate_task.rb:
```diff
@@ -0,0 +1,11 @@
+  end
+
+  def self.org(user)
+    JudgeTeam.for_judge(user.reload)
```
We're assuming here (and elsewhere) that `user` will be a judge who's an admin on a `JudgeTeam`, but it looks like [we get possible users by fetching a list of judges from VACOLS](https://github.com/department-of-veterans-affairs/caseflow/blob/jc/12793-mtv-straight-vacate/app/repositories/task_action_repository.rb#L155). These two aren't always the same, as explained in #8436. It's out of scope for this PR to fix the issue, but maybe we could have a more deliberate error for cases where `JudgeTeam.for_judge(user.reload)` returns `nil`. We'll get an error anyway, because the new task [will fail validation](https://github.com/department-of-veterans-affairs/caseflow/blob/jc/12793-mtv-straight-vacate/app/models/task.rb#L24), but a custom error that highlights the issue here might be useful.

jcq replied:
I added some code and a new `NonexistantJudgeTeam` error; let me know if you think that helps with this issue.

jcq commented on spec/feature/queue/motion_to_vacate_spec.rb:
```diff
@@ -182,10 +182,44 @@
+      expect(new_task.available_actions(motions_attorney)).to include(
+        Constants.TASK_ACTIONS.LIT_SUPPORT_PULAC_CERULLO.to_h
+      )
+      expect(new_task.instructions.join("")).to eq(instructions)
```
It seems like we could navigate back in to the case details page and check the task type, but @leikkisa and I thought it probably makes more sense to do the verification via the DB records since it's faster, and we could test the display of task types in some future PR that might be more design-focused.

jcq commented:
@tomas-nava Alrighty, I think I addressed all of the comments in replies. I also updated the schema to change the comment text — that was a good flag for completeness.

tomas-nava commented on lib/caseflow/error.rb:
```diff
@@ -203,6 +203,12 @@ def initialize(args)
     end
   end
 
+  class NonexistantJudgeTeam < StandardError
```
I'm sorry to report that it's spelled `Nonexistent`

jcq replied:
Yikes, that's an embarrassing typo. Glad you noticed it!

tomas-nava commented on spec/controllers/post_decision_motions_controller_spec.rb:
```diff
@@ -58,7 +61,7 @@ def create_task_without_unnecessary_models
+      assigned_to: judge,
       assigned_by: assigned_by
     )
   end
```
Could we get a small test that expects the `NonexistentJudgeTeam` error?

jcq replied:
Yup!
